### DISCUSSION TITLE
Issues installing for osx-arm64

### DISCUSSION THREAD

**Discussion by m-balcewicz** [PMEAL/porespy#784] [Q&A] [answered]
Hello everyone,

I tried installing porespy for osx-arm64 and got the error of unfound packages. Can somebody comment if providing those packages for the new M1 macs is planned or if I need to use another channel?

In the following, you will find my terminal output:

```
>>> conda install -c conda-forge porespy

Retrieving notices: ...working... done
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.

PackagesNotFoundError: The following packages are not available from current channels:

  - porespy

Current channels:

  - https://conda.anaconda.org/conda-forge/osx-arm64
  - https://conda.anaconda.org/conda-forge/noarch
  - https://repo.anaconda.com/pkgs/main/osx-arm64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/osx-arm64
  - https://repo.anaconda.com/pkgs/r/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.
```

**Comment by vasudevanv** [comment]
I had added porespy and openpnm to the conda-forge-pinning-feedstock repository for migration to osx-arm64. Waiting for the review from the core team.

https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/3918

**Reply by jgostick** [reply]
Hi @vasudevanv, I am sorry but I have literally no idea what you're talking about! What is a pinning feedstock and how does it solve the OP's issue?

**Reply by vasudevanv** [reply]
The good folks at conda-forge automated the migration of packages to the new Apple Silicon architecture through this repository and provided instructions. I just made a PR according to their instructions in the link below

https://conda-forge.org/blog/posts/2020-10-29-macos-arm64/

Once it gets reviewed and merged, conda-forge's pipeline will automatically build packages for the new platform.  I could have alternately sent a PR to https://github.com/conda-forge/porespy-feedstock implementing all the changes manually

**Reply by vasudevanv** [reply]
Let me know if you want me to submit a PR to the porespy-feedstock repo with the addition of the new platform. I think it would hasten the process of resolving the OPs issue.

**Reply by jgostick** [reply]
I think I would rather have this on the porespy-feedstock repo so things are all in 1 place, but I have no experience with this.  There is also the issue that some of the CI tests failed on that PR.

**Reply by vasudevanv** [reply]
So I have the changes for the porespy-feedstock ready. But in my testing the build fails because scikit-fmm and edt have not yet been migrated to the osx-arm64 platform. There are PRs in the feedstocks for those two packages already open for the migration. So we just have to wait until those are merged. Sorry folks :(

**Reply by m-balcewicz** [reply]
Thank you anyway for taking care. Maybe you can leave a short comment when the migration is done? I wish you all the best.

**Comment by m-balcewicz** [comment]
**Update**: Unsure if the migration is done, but I could install the packages today via PyCharm.  

**Update-2**: Although I could install the packages via  PyCharm's library manager, after a short test, I noticed that not all packages had been installed properly. A simple test with the command: "conda install -c conda-forge porespy" shows that my problem is still unresolved. The packages are not yet available for osx-arm64. Sorry for the hasty update - the joy was too big initially.
PS: I Hope to be able to use the library on macOS soon.

**Reply by vasudevanv** [reply]
scikit-fmm which is a dependency currently has no build for osx-arm64. I submitted a PR to their repository for including the osx-arm64 build. Hopefully that should get us a step closer once its merged.

**Reply by jgostick** [reply]
I thought we'd removed skfmm as a hard dependency?  It is a tempermental package, so we only attempt to import within the few functions that use it rather, and report a ModuleNotFoundError if necessary.

**Reply by vasudevanv** [reply]
Unfortunately conda-forge was still picking it up as a dependency. Good news is scikit-fmm developer was very responsive and now we have an osx-arm64 build for it. I have a PR to conda-forge-pinning-feedstock for porespy and openpnm. Once that is merged, we should have osx-arm64 builds available. I manually built the conda package locally and it works!

**Reply by ma-sadeghi** [reply, marked as answer]
@m-balcewicz PoreSpy is now natively [available](https://github.com/conda-forge/porespy-feedstock/pull/17) on ARM64, so you should be able to install it now using `conda install -c conda-forge porespy`.

@vasudevanv Thanks for setting up the groundwork.

**Reply by m-balcewicz** [reply]
Great! Thank you all for your support. The installation has succeeded.
